test/WebSites/OAuth2Integration/AuthServer/Controllers/ConsentController.cs

```diff
@@ -12,46 +12,50 @@ namespace OAuth2Integration.AuthServer.Controllers
     public class ConsentController : Controller
     {
         private readonly IIdentityServerInteractionService _interaction;
-        private readonly IClientStore _clientStore;
-        private readonly IResourceStore _resourceStore;
 
         public ConsentController(
-            IIdentityServerInteractionService interaction,
-            IClientStore clientStore,
-            IResourceStore resourceStore)
+            IIdentityServerInteractionService interaction)
         {
             _interaction = interaction;
-            _clientStore = clientStore;
-            _resourceStore = resourceStore;
         }
 
         [HttpGet("consent")]
         public async Task<IActionResult> Consent(string returnUrl)
         {
             var request = await _interaction.GetAuthorizationContextAsync(returnUrl);
-            var client = await _clientStore.FindEnabledClientByIdAsync(request.ClientId);
-            var resource = await _resourceStore.FindApiResourceAsync("api");
 
             var viewModel = new ConsentViewModel
             {
                 ReturnUrl = returnUrl,
-                ClientName = client.ClientName,
-                ScopesRequested = resource.Scopes.Where(s => request.ScopesRequested.Contains(s.Name))
+                ClientName = request.Client.ClientName,
+                ScopesRequested = request.ValidatedResources?.Resources?.ApiScopes ?? new List<ApiScope>()
```

Comment:
Could this be simplified to this?
```suggestion
                ScopesRequested = request.ValidatedResources?.Resources?.ApiScopes ?? []
```